src/main/kotlin/org/rust/lang/core/resolve/ref/RsLitExprReferenceContributor.kt

```diff
@@ -62,3 +74,60 @@ private class RsLiteralFileReferenceSet(
         }
     }
 }
+
+private val pathValueLiteral: PsiElementPattern.Capture<RsLitExpr> = psiElement<RsLitExpr>()
+    .with("onPathLiteral") { expr -> expr.kind is RsLiteralKind.String && isPathLitExpr(expr) }
+
+private fun isPathLitExpr(expr: RsLitExpr): Boolean {
+    val (function, arguments) = getFunctionAndArguments(expr) ?: return false
+    val owner = ((function.owner as? RsAbstractableOwner.Impl)?.impl?.typeReference?.type as? TyAdt)?.item
+    val (implLookup, knownItems) = expr.implLookupAndKnownItems
+    val isCallExpr = arguments.parent is RsCallExpr
+
+    return when {
+        // Path::new(<literal>)
+        owner == knownItems.Path && function.name == "new" -> true
+        // PathBuf::from(<literal>)
+        owner == knownItems.PathBuf && function.name == "from" -> true
+        // fn foo<P: AsRef<Path>>(p: P) -> foo(<literal>)
+        // or fn foo(p: impl AsRef<Path>) -> foo(<literal>)
+        else -> {
+            var argumentIndex = arguments.exprList.indexOf(expr)
+            if (argumentIndex == -1) return false
+
+            // UFCS, skip self argument
+            if (function.hasSelfParameters && isCallExpr && argumentIndex > 0) {
```

Comment:
Why we need `argumentIndex > 0` condition?
It seems it produces wrong completion in the following case:
```rust
use std::path::Path;

struct Foo;

impl Foo {
    fn foo<T : AsRef<Path>>(&self, path: T) {}
}

fn main() {
    Foo::foo("<caret>"); // I don't expect completion here 
}
```